bluemix/configuration/config_helpers/helpers_test.go

```diff
@@ -10,6 +20,59 @@ package config_helpers
 // 	"github.ibm.com/bluemix-cli-release/build/src/github.ibm.com/Bluemix/bluemix-cli-common/file_helpers"
 // )
 
+func captureEnv() []string {
+	return os.Environ()
+}
+
+func resetEnv(env []string) {
+	os.Clearenv()
+	for _, e := range env {
+		pair := strings.Split(e, "=")
+		os.Setenv(pair[0], pair[1])
+	}
+}
+
+func TestConfigDir(t *testing.T) {
+	assert := assert.New(t)
+
+	env := captureEnv()
+	defer resetEnv(env)
+
+	userHome, err := ioutil.TempDir("", "config_dir_test")
+	assert.NoError(err)
+
+	os.Unsetenv("IBMCLOUD_CONFIG_HOME")
+	os.Unsetenv("IBMCLOUD_HOME")
+	os.Unsetenv("BLUEMIX_HOME")
+	os.Setenv("HOME", userHome)
+
+	assert.NoError(os.RemoveAll(userHome))
+
+	// directory should not exist - will use bluemix
+	assert.Equal(filepath.Join(userHome, ".bluemix"), ConfigDir())
+
+	// create a .ibmcloud directory and it should be returned
+	ibmcloudDir := filepath.Join(userHome, ".ibmcloud")
+	assert.NoError(os.MkdirAll(ibmcloudDir, 0755))
+	assert.Equal(ibmcloudDir, ConfigDir())
+
+	// if only BLUEMIX_HOME is set, BLUEMIX_HOME is used
+	os.Setenv("BLUEMIX_HOME", "/my_bluemix_home")
+	assert.Equal(filepath.Join("/my_bluemix_home", ".bluemix"), ConfigDir())
+
+	// if BLUEMIX_HOME and IBMCLOUD_HOME is set, IBMCLOUD_HOME is used
+	os.Setenv("IBMCLOUD_HOME", "/my_ibmcloud_home")
+	assert.Equal(filepath.Join("/my_ibmcloud_home", ".bluemix"), ConfigDir())
+
+	// if IBMCLOUD_CONFIG_HOME is set but does not exist, IBMCLOUD_HOME is used
+	os.Setenv("IBMCLOUD_CONFIG_HOME", "/my_ibmcloud_config_home")
+	assert.Equal(filepath.Join("/my_ibmcloud_home", ".bluemix"), ConfigDir())
+
+	// if IBMCLOUD_CONFIG_HOME is set and exists, IBMCLOUD_CONFIG_HOME is used
+	os.Setenv("IBMCLOUD_CONFIG_HOME", userHome)
+	assert.Equal(userHome, ConfigDir())
+}
+
 // func TestMigrateFromOldConfig(t *testing.T) {
```

Comment:
The method TestConfigDir() covers many cases, better to put each in a separate test method.

Reply:
Test cases separated.